File: openrndr-jvm/openrndr-adopted-artifex/src/main/java/io/lacuna/artifex/Polar2.java
package io.lacuna.artifex;

/**
 * @author ztellman
 */
public class Polar2 {
  public final double theta;
  public final double r;

  public Polar2(double theta, double r) {
    this.theta = theta;
    this.r = r;
  }

  public Polar2 rotate(double theta) {
    return new Polar2(this.theta + theta, r);
  }

  public Vec2 vec2() {
    double x = Math.cos(theta);
    double y = Math.sin(theta);
    return new Vec2(x * r, y * r);
  }
}

File: openrndr-kartifex/src/commonMain/kotlin/Polar2.kt
package org.openrndr.kartifex

import kotlin.jvm.JvmRecord
import kotlin.math.cos
import kotlin.math.sin

@JvmRecord
data class Polar2(val theta: Double, val r: Double) {
    fun rotate(theta: Double) = Polar2(this.theta + theta, r)

    fun vec2(): Vec2 {
        val x = cos(theta)
        val y = sin(theta)
        return Vec2(x * r, y * r)
    }
}
